src/omega_edit/omega_edit.js

```diff
@@ -88,6 +85,32 @@ function saveSession() {
   })
 }
 
+function replace(data, offset, len) {
+  vscode.postMessage({
+    command: 'replace',
+    data: data,
+    offset: offset,
+    len: len,
+  })
+}
+
+function searchSession(searchPattern, caseInsensitive) {
+  vscode.postMessage({
+    command: 'search',
+    searchPattern: searchPattern,
+    caseInsensitive: caseInsensitive,
+  })
+}
+
+function searchAndReplace(searchPattern, replaceText, caseInsensitive) {
```

Comment:
I'm not sure this makes sense, but .... So is this text only, or is this used for binary/bytes data also with the replacement text being hex?

I guess I'm looking for a way to keep straight what's for binary vs. text only vs. are there things used in multiple modes?

Reply:
So currently we are only searching and replacing using text in the UI. Currently we only have the editing of text fully implemented in the UI. In the future we plan to support both editing the binary data as well, this is already support by omega-edit just needs implemented in the frontend, but for now mostly focus on text.